sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -188,7 +188,7 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
   private def verifySchema(schema: StructType): Unit = {
     schema.foreach { field =>
       field.dataType match {
-        case _: ArrayType | _: MapType | _: StructType =>
+        case _: ArrayType | _: MapType | _: StructType | NullType | CalendarIntervalType =>
```

Comment:
A more general comment. This uses blacklisting, this is problematic when we introduce a `DataType` which should not be written to CSV. Whitelisting is preferred because it is more defensive (it would fail in the mentioned scenario).

How does this work with UDTs?